jupyter_events/logger.py

```diff
@@ -107,94 +79,48 @@ def get_handlers():
         eventlogger_cfg = Config({"EventLogger": my_cfg})
         super()._load_config(eventlogger_cfg, section_names=None, traits=None)
 
-    def register_schema_file(self, filename):
-        """
-        Convenience function for registering a JSON schema from a filepath
-
-        Supports both JSON & YAML files.
+    def register_event_schema(self, schema: Union[dict, str]):
+        """Register this schema with the schema registry.
 
-        Parameters
-        ----------
-        filename: str, path object or file-like object
-            Path to the schema file or a file object to register.
+        Get this registered schema using the EventLogger.schema.get() method.
         """
-        # Just use YAML loader for everything, since all valid JSON is valid YAML
+        self.schemas.register(schema)
 
-        # check if input is a file-like object
-        if hasattr(filename, "read") and hasattr(filename, "write"):
-            self.register_schema(yaml.load(filename))
-        else:
-            with open(filename) as f:
-                self.register_schema(yaml.load(f))
-
-    def register_schema(self, schema):
-        """
-        Register a given JSON Schema with this event emitter
+    def register_event_schema_file(self, schema_file: str):
+        """Register this schema with the schema registry.
```

Comment:
Should probably indicate that this method takes a file path instead of the schema itself.

Reply:
I've consolidated the `register_event_schema_file` with `register_event_schema`. That method can now receive either a string, pathlib object, or schema object.